azure-pipelines/publish-codecoverage.yml

```diff
@@ -16,20 +16,12 @@ steps:
   continueOnError: true
   condition: ${{ parameters.includeMacOS }}
 - powershell: |
-    dotnet tool install --tool-path obj dotnet-reportgenerator-globaltool --version 4.8.5 --configfile azure-pipelines/justnugetorg.nuget.config
-    Copy-Item -Recurse $(Pipeline.Workspace)/coverageResults-Windows/obj/* $(System.DefaultWorkingDirectory)/obj
-    Write-Host 'Substituting {reporoot} with $(System.DefaultWorkingDirectory)'
-    $reports = Get-ChildItem -Recurse '$(Pipeline.Workspace)/coverage.*cobertura.xml'
-    $reports |% {
-        $content = Get-Content -Path $_ |% { $_.Replace('{reporoot}', '$(System.DefaultWorkingDirectory)') }
-        Set-Content -Path $_ -Value $content -Encoding UTF8
-    }
```

Comment:
@marcorossignoli I would replace machine-specific source file paths with `{reporoot}`, on each agent that ran tests, then here I would replace it 'back' to a machine path that matches *this* agent. I would also place build-generated source files back on disk where they were expected, all so that coverage reports could display those source files. Do I still need to do this by hand, or does `dotnet-coverage merge` somehow magically take care of it?

Reply:
I have to keep the `{reporoot}` substitution in order for the reports to merge into one that refers to each file once, and to be able to find the source files while it creates the html report shown by AzP.